src/assets/Azure.Core.Shared/Argument.cs

```diff
@@ -212,5 +212,24 @@ public static string CheckNotNullOrEmpty(string value, string name)
             AssertNotNullOrEmpty(value, name);
             return value;
         }
+
+        public static void AssertHasOnlySupportedHeaders(RequestConditions? requestConditions, string paramName, string[] supportedHeaders)
```

Comment:
I believe files under Azure.Core.Shared are copied in from the azure-sdk-for-net repo, so this change may need to be made there.

Reply:
Yes, we need to update in azure-sdk-for-net repo, after we merge those changes into azure-sdk-for-net repo, we need first file an PR in autorest.csharp to update the Argument.cs which is copied from azure-sdk-for-net. Then file another PR to include our business logic. We can merge those two PR as one.